lab_5_scrapper/scrapper.py

```diff
@@ -2,73 +2,172 @@
+    """
+    pass
+
+
+class IncorrectEncodingError(Exception):
+    """
+    Raises when encoding does not specified as a string
+    """
+    pass
+
+
+class IncorrectTimeoutError(Exception):
+    """
+    Raises when timeout value is not a positive integer less than 60
+    """
+    pass
+
+
+class IncorrectVerifyError(Exception):
+    """
+    Raises when verify certificate value does not either be True or False
+    """
+    pass
 
 
 class Config:
     """
     Unpacks and validates configurations
     """
-
     def __init__(self, path_to_config: Path) -> None:
         """
         Initializes an instance of the Config class
         """
-        pass
+        self.path_to_config = path_to_config
+        self._config_data = self._extract_config_content()
+        self._validate_config_content()
+        self._seed_urls = self._config_data.seed_urls
+        self._num_articles = self._config_data.total_articles_to_find_and_parse
+        self._headers = self._config_data.headers
+        self._encoding = self._config_data.encoding
+        self._timeout = self._config_data.timeout
+        self._verify_certificate = self._config_data.should_verify_certificate
+        self._headless_mode = self._config_data.headless_mode
 
     def _extract_config_content(self) -> ConfigDTO:
         """
         Returns config values
         """
-        pass
+        with open(self.path_to_config, 'r', encoding='utf-8') as f:
+            content = json.load(f)
+        return ConfigDTO(**content)
 
     def _validate_config_content(self) -> None:
         """
         Ensure configuration parameters
         are not corrupt
         """
-        pass
+        with open(self.path_to_config, 'r', encoding='utf-8') as f:
+            content = json.load(f)
+
+        seed_urls = content['seed_urls']
+        headers = content['headers']
+        total_articles_to_find_and_parse = content['total_articles_to_find_and_parse']
+        encoding = content['encoding']
+        timeout = content['timeout']
+        should_verify_certificate = content['should_verify_certificate']
+
+        if not isinstance(seed_urls, list)\
+                or not not all(isinstance(url, str) for url in seed_urls)\
+                or not all(re.match('https?://w?w?w?.', url) for url in seed_urls):
+            raise IncorrectSeedURLError
+
+        if not isinstance(headers, dict):
+            raise IncorrectHeadersError
+
+        if not isinstance(encoding, str):
+            raise IncorrectEncodingError
+
+        if not isinstance(total_articles_to_find_and_parse, int):
+            raise IncorrectNumberOfArticlesError
+
+        if total_articles_to_find_and_parse not in range(1, 151):
+            raise NumberOfArticlesOutOfRangeError
+
+        if not isinstance(timeout, int) or not timeout < 60 or not timeout > 0:
+            raise IncorrectTimeoutError
+
+        if not isinstance(should_verify_certificate, bool):
+            raise IncorrectVerifyError
```

Comment:
what about headless mode validation?

Reply:
made a new class for this type of Error and added new check